### PR TITLE
Fix Speed Controller vs Motor Controller sitewide

### LINKED ISSUE #1002: Fix Speed Controller vs Motor Controller sitewide
Many documents are split on the terminology.  Will try to include notes indicating their equivalence but since first calls them all Motor Controllers in the game manual we should too!

### PR DESCRIPTION
**Pull request by jasondaming**
Closes #1002

### CODE REVIEW

File: source/docs/software/actuators/overview.rst [hunk 1]
```diff
@@ -2,20 +2,21 @@
 
 Actuator Overview
 ==================
-This section discusses the control of motors and pneumatics through speed controllers, solenoids and pneumatics, and their interface with C++ and Java WPILib.
+This section discusses the control of motors and pneumatics through motor controllers, solenoids and pneumatics, and their interface with C++ and Java WPILib.
```
Comment (sciencewhiz) [inline source/docs/software/actuators/overview.rst:5]: This sentence doesn't make sense. Pneumatics are controlled by pneumatics?

File: source/docs/software/actuators/wpi-drive-classes.rst [hunk 2]
```diff
@@ -46,11 +46,11 @@ Motor Safety
...
+The purpose of the Motor Safety mechanism is the same as the purpose of a watchdog timer, to disable mechanisms which may cause harm to themselves, people or property if the code locks up and does not properly update the actuator output. Motor Safety breaks this concept out on a per actuator basis so that you can appropriately determine where it is necessary and where it is not. Examples of mechanisms that should have motor safety enabled are systems like drive trains and arms. If these systems get latched on a particular value they could cause damage to their environment or themselves. An example of a mechanism that may not need motor safety is a spinning flywheel for a shooter. If this mechanism gets latched on a particular value it will simply continue spinning until the robot is disabled. By default Motor Safety is enabled for RobotDrive objects and disabled for all other motor controllers and servos.
```
Comment (sciencewhiz) [inline source/docs/software/actuators/wpi-drive-classes.rst:49]: Motor safety is also enabled for differential drive and probably the other drivetrain classes.

File: source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst [hunk 3]
```diff
@@ -30,13 +30,13 @@ To specify the correct settings for your PID controller, use the following optio
...
+- **Controller Type:** This specifies whether the controller is an onboard RIO loop, or is running on a smart motor controller such as a Talon or a SPARK MAX.
 - **Post-Encoder Gearing:** This specifies the gearing between the encoder and the mechanism itself. This is necessary for control loops that do not allow user-specified unit scaling in their PID computations (e.g. those running on Talons). This will be disabled if not relevant.
 - **Encoder EPR:** This specifies the edges-per-revolution (not cycles per revolution) of the encoder used, which is needed in the same cases as Post-Encoder Gearing.
 - **Has Follower:** Whether there is a motor controller following the controller running the control loop, if the control loop is being run on a peripheral device. This changes the effective loop period.
-- **Follower Update Period:** The rate at which the follower (if present) is updated. By default, this is 100Hz (every 0.01s) for Talons and Sparks, but can be changed.
+- **Follower Update Period:** The rate at which the follower (if present) is updated. By default, this is 100Hz (every 0.01s) for the Talon and the SPARK, but can be changed.
```
Comment (sciencewhiz) [inline source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst:37]: I think this should specify Talon SRX, Falcon, and Spark Max. Non-CAN talons don't have followers.

File: source/docs/software/actuators/overview.rst (hunk 1 above)
Comment (jasondaming) [reply source/docs/software/actuators/overview.rst:5]: What about if we just remove the "through...." so it is: This section discusses the control of motors and pneumatics and their interface with C++ and Java in WPILib.

File: source/docs/software/actuators/wpi-drive-classes.rst (hunk 2 above)
Comment (jasondaming) [reply source/docs/software/actuators/wpi-drive-classes.rst:49]: If we know what classes (I don't know of any other that Diff Drivetrain) then I will update this

Comment (sciencewhiz) [reply source/docs/software/actuators/wpi-drive-classes.rst:49]: DifferentialDrive, KilloughDrive, MecanumDrive

**Comment by AustinShalit** [comment]
I am still unsure of this change because we call them SpeedControllers in code.

**Comment by jasondaming** [comment]
The places where the SpeedController Class is referenced were not changed.  In the end I don't see how we can go against the FRC manual isn't that our "single source of truth"?

If anything I think the case should be made for changing the code.  That is such a breaking change that I don't think it is worth it but this conflicting terminology doesn't help anyone.

**Comment by Daltz333** [comment]
@PeterJohnson what are your thoughts on this

**Comment by Daltz333** [comment]
It was determined that motor controller is the right terminology.

**Comment by jasondaming** [comment]
Just fixed the merge conflicts let me know if there is anything else for this one

**Review by AustinShalit — DISMISSED** [review]

File: source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst [hunk 4]
```diff
@@ -30,13 +30,13 @@ To specify the correct settings for your PID controller, use the following optio
...
+- **Controller Type:** This specifies whether the controller is an onboard RIO loop, or is running on a smart motor controller such as a Talon or a SPARK MAX.
 - **Post-Encoder Gearing:** This specifies the gearing between the encoder and the mechanism itself. This is necessary for control loops that do not allow user-specified unit scaling in their PID computations (e.g. those running on Talons). This will be disabled if not relevant.
 - **Encoder EPR:** This specifies the edges-per-revolution (not cycles per revolution) of the encoder used, which is needed in the same cases as Post-Encoder Gearing.
 - **Has Follower:** Whether there is a motor controller following the controller running the control loop, if the control loop is being run on a peripheral device. This changes the effective loop period.
-- **Follower Update Period:** The rate at which the follower (if present) is updated. By default, this is 100Hz (every 0.01s) for Talons and Sparks, but can be changed.
+- **Follower Update Period:** The rate at which the follower (if present) is updated. By default, this is 100Hz (every 0.01s) for the Talon SRX, Talon FX and the SPARK MAX, but can be changed.
```
Comment (AustinShalit) [inline source/docs/software/wpilib-tools/robot-characterization/analyzing-feedback.rst:37]:
```suggestion
- **Follower Update Period:** The rate at which the follower (if present) is updated. By default, this is 100Hz (every 0.01s) for the Talon SRX, Talon FX, and the SPARK MAX, but can be changed.
```

**Pull request merged**